// @uys2000/u-http/src/service/http.mjs
export const getUrl = (url, params = {}) => {
  const uri = new URL(url);
  Object.keys(params ?? {}).forEach((key) => {
    if (!params[key]) return;
    uri.searchParams.set(key, params[key]);
  });
  return uri.toString();
};
export const doRequest = async (url, method, params, body, headers = {}) => {
  return window.fetch(getUrl(url, params), {
    headers: headers ? { ...headers } : {},
    method: method,
    body: body ? JSON.stringify(body) : undefined,
  });
};

export const doHead = async (url, params, headers) => {
  return doRequest(url, "HEAD", params, undefined, headers);
};
export const doGet = async (url, params, headers) => {
  return doRequest(url, "GET", params, undefined, headers);
};
export const doPost = async (url, params, body, headers) => {
  return doRequest(url, "POST", params, body, headers);
};
export const doPut = async (url, params, body, headers) => {
  return doRequest(url, "PUT", params, body, headers);
};
export const doPatch = async (url, params, body, headers) => {
  return doRequest(url, "PATCH", params, body, headers);
};
export const doDelete = async (url, params, body, headers) => {
  return doRequest(url, "DELETE", params, body, headers);
};
